roles/matrix-bridge-appservice-slack/defaults/main.yml

```diff
@@ -0,0 +1,92 @@
+matrix_appservice_slack_matrix_port: 8008
+matrix_appservice_slack_slack_port: 9003
+
+matrix_appservice_slack_homeserver_address: 'http://matrix-synapse:{{ matrix_appservice_slack_matrix_port }}'
+matrix_appservice_slack_homeserver_domain: '{{ matrix_domain }}'
+matrix_appservice_slack_appservice_address: 'http://matrix-appservice-slack:{{ matrix_appservice_slack_matrix_port }}'
+
+# A list of extra arguments to pass to the container
+matrix_appservice_slack_container_extra_arguments: []
+
+# List of systemd services that matrix-appservice-slack.service depends on.
+matrix_appservice_slack_systemd_required_services_list: ['docker.service']
+
+# List of systemd services that matrix-appservice-slack.service wants
+matrix_appservice_slack_systemd_wanted_services_list: []
+
+matrix_appservice_slack_appservice_token: ''
+matrix_appservice_slack_homeserver_token: ''
+matrix_appservice_slack_id_token: ''
+
+matrix_appservice_slack_configuration_yaml: |
+    slack_hook_port: {{ matrix_appservice_slack_slack_port }}
+    #{{matrix_homeserver_url}} had been replaced with {{matrix_domain}} hmmmm....
+    inbound_uri_prefix: "{{ matrix_homeserver_url }}:{{ matrix_appservice_slack_slack_port }}"
+    bot_username: "{{ matrix_appservice_slack_bot_name }}"
+    username_prefix: "slackuser_"
+
+    homeserver:
+        url: "{{ matrix_homeserver_url }}:{{matrix_appservice_slack_matrix_port}}"
+        server_name: "{{ matrix_domain }}"
+
+    matrix_admin_room: "{{ slack_control_room_id }}"
+
+
+matrix_appservice_slack_configuration_extension_yaml: |
+    #slack_hook_port: 9898
+    #inbound_uri_prefix: "https://my.server.here:9898/"
+    #bot_username: "slackbot"
+    #username_prefix: "slack_"
+    # Optional
+    #slack_master_token: "abc-123-def"
+    # Optional
+    #matrix_admin_room: "!aBcDeF:matrix.org"
+    #homeserver:
+    #    url: http://localhost:8008
+    #    server_name: my.server
+    # Optional
+    #tls:
+    #    key_file: /path/to/tls.key
+    #    crt_file: /path/to/tls.crt
+    #logging:
+    #    console: "info"
+    #    files:
+    #    - "./debug.log": "info"
+    #- "./error.log": "error"
+
+matrix_appservice_slack_configuration_extension: "{{ matrix_appservice_slack_configuration_extension_yaml|from_yaml if matrix_appservice_slack_configuration_extension_yaml|from_yaml else {} }}"
+
+matrix_appservice_slack_configuration: "{{ matrix_appservice_slack_configuration_yaml|from_yaml|combine(matrix_appservice_slack_configuration_extension, recursive=True) }}"
+
+matrix_appservice_slack_registration_yaml: |
+  id: "{{ matrix_appservice_slack_id_token }}"
+  as_token: "{{ matrix_appservice_slack_appservice_token }}"
+  hs_token: "{{ matrix_appservice_slack_homeserver_token }}"
+  namespaces:
+    users:
+    - exclusive: true
+      regex: '@slack_.*'
+    aliases:
+    - exclusive: false
+      regex: '#slack_.*'
+    rooms: []
+  url: "https://localhost:{{matrix_appservice_slack_matrix_port}}"
```

Comment:
What you have (`https://localhost:{{matrix_appservice_slack_matrix_port}}`) can't work, because:
- `localhost`, as interpreted by the Synapse container means "our own (Synapse) container".. which is wrong.
- even if you make it reach the Slack container, `9003` is not an HTTPS port, so using `https://` is not okay

-----------

I would introduce a new variable for this:

```
matrix_appsevice_slack_appservice_url: http://matrix-appservice-slack:9003
```

This makes it possible to override it (if necessary).
Also, the new default value works better with containers (inside the container network).